src/node.js

```diff
@@ -388,6 +396,53 @@
       nextTickQueue.push(obj);
       tickInfo[kLength]++;
     }
+
+    // Internal callbacks that must be run when there is no more micro-tasks
+    // scheduled for this turn.
+    function lateTick(callback) {
+      lateTickQueue.push(callback);
+      scheduleMicrotasks();
```

Comment:
@vkurchatkin `lateTick` is only called when there is an unhandled rejection, it is completely unrelated to whether microtasks have been scheduler or not. In fact you can comment out the line and not a single test passes.